src/Extensions/ElementalAreasExtension.php

```diff
@@ -278,17 +280,21 @@ public function supportsElemental()
      */
     public function ensureElementalAreasExist($elementalAreaRelations)
     {
-        foreach ($elementalAreaRelations as $eaRelationship) {
-            $areaID = $eaRelationship . 'ID';
-
-            if (!$this->owner->$areaID) {
-                $area = ElementalArea::create();
-                $area->OwnerClassName = get_class($this->owner);
-                $area->write();
-                $this->owner->$areaID = $area->ID;
+        $owner = $this->getOwner();
+        // Only create the elemental area on draft stage
+        if (Versioned::get_reading_mode() === 'Stage.' . Versioned::LIVE) {
```

Comment:
```suggestion
        if (Versioned::get_stage() !== Versioned::DRAFT) {
```
Check the stage specifically rather than the whole reading mode, and explicitly check for "not draft" rather than live (e.g. `null` might be returned if we're in archived reading mode)

Reply:
If it's null, that might represent "archived" reading mode which is not draft stage and should not be used to create an elemental area... can you please confirm that using the "archived" reading mode won't result in a new orphaned elemental area?